### DISCUSSION TITLE
BuildQualityChecks@9 warningTaskSelectors problems / filtering tasks / regex

### DISCUSSION THREAD

**Discussion by oatsoda** [MicrosoftPremier/VstsExtensions#227] [General]
Hi

I'm having some difficulties with the `warningTaskSelectors` option.

Everything works fine when I add the BQC to my pipeline, but I there are some Tasks on it which I don't want included in the analysis because those tasks are conditional and only run on the `main` branch - meaning the baseline `main` warnings count is naturally inflated compared to a PR branch pipeline.

To account for this I am trying to limit which Tasks in my pipeline using the `warningTaskSelectors` option.

I tried something simple like `   warningTaskSelectors: '/^(Build|Restore)$'` to match either of the two steps I want to monitor.  However, this gave me the following output:

> [WARNING] The expression '/^(Build|Restore)$' is not a valid regular expression and will be ignored.

I looked back at the [docs](https://marketplace.visualstudio.com/items?itemName=mspremier.BuildQualityChecks) and saw this regex documented, presumably as the default,  `'/^((vs|ms)build|ant(\s+.+)?|gradle(w)?(\s+.+)?|grunt|gulp|maven(\s+.+)?|xamarin(android|ios)|xcode(\s+.+)?|cmake|build\s+.+)$/i'` so I tried setting that value explicitly in my yaml.  This gave me the output:

> [WARNING] The specified task filters did not match any build 

In summary, my questions are:

1. What's wrong with my basic regex `/^(Build|Restore)$` or how can I filter by a simple Task name?
2. What is the default value if I don't specify `warningTaskSelectors` in my pipeline (and therefore what is the one in the docs?)

My pipeline BQC task in full:

```
    - task: BuildQualityChecks@9
      displayName: 'Compiler Warnings Check'
      condition: and(succeeded(), ne(variables['SkipBuildWarningCount'], 'true'))
      inputs:
        checkWarnings: true
        warningFailOption: 'build'
        showStatistics: true        
        includePartiallySucceeded: false
        warningTaskSelectors: '/^(Build|Restore)$'
        baseDefinitionId: 3
        baseBranchRef: 'refs/heads/main'
```

Thanks!

**Comment by oatsoda** [comment]
I have realised that I am missing the trailing `/` on my regex :(

`/^(Build|Restore)$/`
